portal/app/core/views/base.py

```diff
@@ -153,3 +153,11 @@ def charts_no_control(request):
     if request.user.powernetuser.type == PowernetUserType.LAB:
         return render(request, 'partials/chart_plots_no_control.html')
     return render(request, 'partials/404.html')
+
+
+@login_required
+def local_fan_info(request):
+    if request.user.powernetuser.type == PowernetUserType.FARM:
+        return render(request, 'partials/local_fan.html')
+    else:
+        return render(request, 'partials/404.html')
```

Comment:
simpler 👇
```python
if request.user.powernetuser.type == PowernetUserType.FARM:
        return render(request, 'partials/local_fan.html')
return render(request, 'partials/404.html')
```

Also, maybe a 403 instead of 404?

Reply:
yeah makes sense, thanks